### ISSUE TITLE
Porting updates from Athena to standalone tool

### ISSUE THREAD

**Issue by matthewfeickert** [dguest/ttree2hdf5#9]
> This is better maintained within ATLAS Athena releases. Please file an issue if you'd like to use that version stand alone.

@dguest, not that either of us have much time to work on this right now, but as the [HDF5 Utilities](https://gitlab.cern.ch/atlas/athena/tree/21.2/PhysicsAnalysis/AnalysisCommon/HDF5Utils) module in Athena has been more recently updated than the standalone version of the tool here how much work would it take to port over the updates? Did you need to refactor the code much to get it working in Athena? Or can we just copy the code over in a series of PRs?

**Comment by dguest** [comment]
I really would like to port a lot of the work there into an this library. You're not the first person to ask.

I think pulling over the commits would be pretty ugly, though: I had to change the naming conventions, use ATLAS-specific cmake, and move the whole project into a subdirectory. Unfortunately I think this means that any patching is not that straightforward. Do you have any experience maintaining forks where one fork puts the project root directory somewhere else? 

It might be easiest to just start over from the ATLAS version, replacing the ATLAS CMake with something more standard, sadly. I guess porting between versions would still be tricky because of the difference in package root directory I mentioned above.

There was also some discussion on using ATLAS copyright vs the MIT license. In the [original merge request in which I added it to ATLAS](https://gitlab.cern.ch/atlas/athenaprivate1/-/merge_requests/8399) I ended up removing all the MIT license stuff because I didn't feel like dealing with all the bickering that accompanies doing anything remotely unusual within ATLAS. But I don't think there's any problem with releasing the same code under multiple licenses: the only tricky part might be that we should add Jon Burr to the list of contributors, since he added a merging utility.

**Comment by matthewfeickert** [comment]
> Do you have any experience maintaining forks where one fork puts the project root directory somewhere else?

No, as I think most projects try to avoid having these discrepancies.

> It might be easiest to just start over from the ATLAS version, replacing the ATLAS CMake with something more standard, sadly. I guess porting between versions would still be tricky because of the difference in package root directory I mentioned above.

That seems reasonable.

> There was also some discussion on using ATLAS copyright vs the MIT license. In the [original merge request in which I added it to ATLAS](https://gitlab.cern.ch/atlas/athenaprivate1/-/merge_requests/8399) I ended up removing all the MIT license stuff because I didn't feel like dealing with all the bickering that accompanies doing anything remotely unusual within ATLAS. But I don't think there's any problem with releasing the same code under multiple licenses.

Athena doesn't even have a proper license in it. It is currently

> The software in this repository is released under the Apache 2.0 license, except where other licenses apply.

which is not a license (also Apache 2.0 is a **template**). Changing the license back to MIT is already compatible under Apache 2.0, so there isn't an issue here at all. 

>  the only tricky part might be that we should add Jon Burr to the list of contributors, since he added a merging utility.

Also, hi @j0nburr and thanks! :wave: Jon can just be [added to the commit that gets merged in as a co-author on the commit](https://docs.github.com/en/github/committing-changes-to-your-project/creating-a-commit-with-multiple-authors).
